controllers/release/release_adapter.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package release
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	libhandler "github.com/operator-framework/operator-lib/handler"
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/controllers/results"
+	"github.com/redhat-appstudio/release-service/tekton"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"time"
+)
+
+// Adapter holds the objects needed to reconcile a Release.
+type Adapter struct {
+	release *v1alpha1.Release
+	logger  logr.Logger
+	client  client.Client
+	context context.Context
+}
+
+// releaseFinalizer is the finalizer name to be added to the Releases
+const releaseFinalizer string = "appstudio.redhat.com/finalizer"
```

Comment:
Not needed but yeah, good idea. Also, the variable name should be renamed to `finalizer` as it’s already in the `release` package.

Reply:
I know it isn't needed but you did a lot of work to make it "easy" to add more controllers with this style later. If we add another resource with a finalizer, it would suck that this string has nothing to do with the `Release` kind in it. +1 to the variable name